src/client.ts

```diff
@@ -158,20 +164,28 @@ function normalizeConfig(config: ClickHouseClientConfigOptions) {
       }
     }
   }
+
+  const url = config.url && createUrl(config.url)
+
   return {
     application_id: config.application,
-    url: createUrl(config.host ?? 'http://localhost:8123'),
+    url: createUrl(
+      config.host ?? (url && `http://${url.host}`) ?? 'http://localhost:8123'
+    ),
     request_timeout: config.request_timeout ?? 300_000,
     max_open_connections: config.max_open_connections ?? Infinity,
     tls,
     compression: {
       decompress_response: config.compression?.response ?? true,
       compress_request: config.compression?.request ?? false,
     },
-    username: config.username ?? 'default',
-    password: config.password ?? '',
-    database: config.database ?? 'default',
-    clickhouse_settings: config.clickhouse_settings ?? {},
+    username: config.username ?? (url && url.username) ?? 'default',
```

Comment:
I agree that configuration should take precedence, but I'd expect the client to throw an exception to ask an end user to specify the correct value explicitly.

Reply:
I think it should be reflected in documentation, and not throw error cause it's not something that will break execution. It's also very flexible to have, for example following setup: 

```Javascript
const client = createClient({ url: 'clickhouse://john@127.0.0.1:9000', password: process.env.CH_PASSWORD  });
```
It's follows logic beside original CH behaviour (https://github.com/ClickHouse/ClickHouse/pull/50689)